selfdrive/ui/qt/setup/setup.cc

```diff
@@ -170,7 +170,16 @@ QWidget * Setup::network_setup() {
   QPushButton *cont = new QPushButton();
   cont->setObjectName("navBtn");
   cont->setProperty("primary", true);
-  QObject::connect(cont, &QPushButton::clicked, this, &Setup::nextPage);
+  QObject::connect(cont, &QPushButton::clicked, [=]() {
+    QString url = InputDialog::getText(tr("Enter installer URL"), this, "");
+    if (url.isEmpty()) {
+      return;
+    }
+    setCurrentWidget(downloading_widget);
```

Comment:
looks like this is slightly different. the logic here was carefully done to avoid glitches when opening/closing the input popup. did you test that?

Reply:
my bad, I see the problem

Reply:
cool, setup specifically has to be super well tested since we don't use it everyday